runtime/service.go

```diff
@@ -1304,26 +1316,35 @@ func (s *service) Cleanup(requestCtx context.Context) (*taskAPI.DeleteResponse,
 	}, nil
 }
 
-func (s *service) monitorVMExit() {
-	defer func() {
+// cleanup resources
+func (s *service) cleanup() error {
+	s.cleanupOnce.Do(func() {
 		// we ignore the error here due to cleanup will only succeed if the jailing
 		// process was killed via SIGKILL
 		if err := s.jailer.Close(); err != nil {
+			s.vmExitErr = err
```

Comment:
We potentially could be stomping over the error here. How do you feel about just using an error in `cleanup`'s scope?

Reply:
Let me check. This function might be the only one who sets vmExitErr.

Reply:
This function is the only one who uses (read/write) the variable.

Reply:
I've renamed the variable from vmExitErr to cleanupErr, since this is only used by cleanup. We still need to keep it in `*service` to make sure calling cleanup() always returns the same value.